site/src/Consumer/EmailNotificationConsumer.php

```diff
@@ -12,26 +14,107 @@
  */
 class EmailNotificationConsumer extends AbstractNotificationConsumer
 {
+    /**
+     * @var UserManagerInterface
+     */
+    protected $userManager;
+
+    /**
+     * @var \Swift_Mailer
+     */
+    protected $mailer;
+
+    /**
+     * @var Environment
+     */
+    protected $twigEnvironment;
+
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
     /**
      * EmailNotificationConsumer constructor
      *
-     * @param LoggerInterface $logger
+     * @param LoggerInterface      $logger
+     * @param UserManagerInterface $userManager
+     * @param Environment          $twigEnvironment
+     * @param \Swift_Mailer        $mailer
+     * @param ContainerInterface   $container
      */
-    public function __construct(LoggerInterface $logger)
-    {
+    public function __construct(
+        LoggerInterface $logger,
+        UserManagerInterface $userManager,
+        Environment $twigEnvironment,
+        ContainerInterface $container,
+        \Swift_Mailer $mailer
+    ) {
         parent::__construct($logger);
+        $this->userManager = $userManager;
+        $this->twigEnvironment = $twigEnvironment;
+        $this->container = $container;
+        $this->mailer = $mailer;
+    }
+
+    protected function shouldHandle()
+    {
+        $user = $this->userManager->findUserByUsername($this->notification->getRecipient()->getUsername());
+
+        $emailSubscription = $user->isSubscribedToEmailNotifications();
+
+        if (!$emailSubscription) {
+            $this->logger->info('You have unsubscribed to email notifications from LibreCores');
```

Comment:
Either remove this, or make it more useful by including the user name instead of "you".